code/modules/mob/login.dm

```diff
@@ -32,6 +32,7 @@
 	if(!client)
 		return
 
+	canon_client = client
```

Comment:
This was actually missing from a tg port, this should hopefully help with client logout cleanup since it can actually clean up now.